<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary

import pandas as pd

memberData = {
    'name' : ['Shani', 'Eve', 'Freya'],
    'age' : [21, 16, 14],
    'team' : ['KIII', 'J', 'T']
}

df = pd.DataFrame(memberData, index=['1', '2', '3'])

print(df)

name  age  team
1  Shani   21  KIII
2    Eve   16     J
3  Freya   14     T

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [7]:
# Membaca data dan memasukannya kedalam suatu variabel

df = pd.read_csv('diamonds-1.csv')

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

1. Perbedaan Series dengan List dan Dictionary
- Series bisa mempunyai key yang duplikat.
- Series memiliki kemampuan untuk slicing data, tidak seperti Dict.
- Ketika di-print Series menampilkan beserta Index, sedangkan List hanya menampilkan data.
- Index Series tidak harus integer (bisa 'a', 'b', 'c'), sedangkan List harus integer.

2. Perbedaan Dataframe dengan Series
- Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data seperti integer, string, dan lain sebagainya. Tipe data dari objek series ini harus seragam.
- DataFrame adalah suatu objek 2 dimensi tempat menyimpan data dengan lebih terstruktur. Dataframe memiliki 2 index, yaitu index baris dan index columns. Dalam satu column, dataframe harus memiliki tipe data yang sama, tapi antar column, dataframe bisa memiliki jenis data yang berbeda.

In [4]:
# Data Series

data = pd.Series([5, 2, 3, 7], index=['a', 'b', 'c', 'd'])

print(data)

a    5
b    2
c    3
d    7
dtype: int64


<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [8]:
# Baris pertama dari column price

print(df.iloc[0]['price'])

# Baris terakhir dari column color
print(df.iloc[-1]['color'])

326
D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab:

Statistika deskriptif merupakan suatu metode yang berhubungan dengan pengumpulan atau penyajian data sampai memberi informasi yang berguna.

Untuk contoh dari statistika deskriptif sendiri antara lain:
- tabel
- diagram
- grafik
- besaran-besaran lain dalam majalah dan koran-koran.

Dengan menggunakan statistika deskriptif, berbagai kumpulan data bisa tersaji dengan ringkas dan juga rapi serta mampu memberikan informasi inti dari kumpulan data yang ada.

Penyajian data dalam kategori deskriptif bisa berupa grafis dan numerik:

1. Penyajian Data dalam Bentuk Grafis
    - Histogram
    - Pie Chart
    - Poligon
    - Ogive
    - Diagram Batang Daun (Stem and Leaf)
2. Penyajian Data Numerik
    - Central Tredency
    - Dispersion (Pencaran)
    - Fractile
    - Skewness
    - Pengukuran Keruncingan

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [14]:
# code here

print(df['price'].mean())
print(df['depth'].std())
print(df['carat'].max())

3932.799721913237
1.4326213188337733
5.01


Jawab:

Setelah menggunakan metode Pandas, didapat statistik deskriptif:

- Nilai rata-rata dari kolom price sebesar 3932.799721913237.
- Nilai standar deviasi dari kolom depth sebesar 1.4326213188337733.
- Nilai maksimum dari kolom carat sebesar 5.01